File: src/provider/PropertiesProvider.java
package provider;

import com.intellij.psi.PsiElement;
import com.intellij.psi.xml.XmlTag;
import domain.Dependency;

import java.util.Arrays;
import java.util.Optional;
import java.util.stream.Collectors;
import java.util.stream.Stream;

import static domain.constant.XmlNodeName.PROPERTIES;

/**
 *
 */
public class PropertiesProvider {

    /**
     *
     * @param project
     * @return
     */
    public XmlTag provide(final XmlTag project) {
        return Optional.ofNullable(project)
                .map(PsiElement::getChildren)
                .map(this::extractProperties)
                .orElse(null);
    }

    private XmlTag extractProperties(PsiElement[] psiElements) {
        return Arrays.stream(psiElements)
                .filter(XmlTag.class::isInstance)
                .map(XmlTag.class::cast)
                .filter(xmlTag -> PROPERTIES.getXmlName().equals(xmlTag.getName().toLowerCase()))
                .findFirst()
                .orElse(null);
    }
}

File: src/main/kotlin/com/mirkoalicastro/mavenversionrefactor/provider/PropertiesProvider.kt
package com.mirkoalicastro.mavenversionrefactor.provider

import com.intellij.psi.PsiElement
import com.intellij.psi.xml.XmlTag
import com.mirkoalicastro.mavenversionrefactor.domain.constant.XmlNodeName.PROPERTIES

class PropertiesProvider {

    fun provide(project: XmlTag?) = project?.children?.extractProperties()

    private fun Array<PsiElement>.extractProperties() =
            filter { XmlTag::class.java.isInstance(it) }
                    .map { XmlTag::class.java.cast(it) }
                    .firstOrNull { PROPERTIES.xmlName == it.name.toLowerCase() }
}
